File: C++/2501. Longest Square Streak in an Array.cpp
class Solution {
public:
    int longestSquareStreak(vector<int>& nums) {
        set<int>n(nums.begin(),nums.end());
        int m=0,k;
        long t;
        for(auto a:n){
            k=0;
            t=a;
            while(n.find(t)!=n.end()){
                k++;
                if (t > INT_MAX / t) break;  
                t = t * t;
            }
            if(k>m)
            m=k;
        }
        if(m==1)
        return -1;
        return m;
    }
};

File: JAVA/2501. Longest Square Streak in an Array.java
class Solution {
    public int longestSquareStreak(int[] nums) {
        Set<Integer>n=new HashSet<>();
        for(int i:nums){
            n.add(i);
        }
        int m=0,k,t;
        for(Integer i:n){
            t=i;
            k=0;
            while(n.contains(t)){
                k++;
                if(t>Integer.MAX_VALUE/t)
                break;
                t=t*t;
            }
            if(k>m)
            m=k;
        }
        if(m==1)
        return -1;
        return m;
    }
}